src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md

```diff
@@ -16,4 +15,35 @@ This can be done by:
     <AdditionalFiles Include="PublicAPI.Shipped.txt" />
     <AdditionalFiles Include="PublicAPI.Unshipped.txt" />
   </ItemGroup>
-```
\ No newline at end of file
+```
+
+## Conditional API Differences
+
+Sometimes APIs vary by compilation symbol such as target framework.
+
+For example when using the [`#if` preprocessor directive](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-if):
+
+```c#
+        public void Foo(string s)
+        {}
+
+#if NETCOREAPP3_0
+        public void Foo(ReadOnlySpan<char> s)
+        {}
+#else
+```
+
+To correctly model the API differences between target frameworks (or any other property), use multiple instances of the `PublicAPI.*.txt` files.
+
+For example, if you target both `net4.8` and `netcoreapp3.0` target frameworks, and APIs differ between each, then you would have the following:
+
+```xml
+  <ItemGroup Condition="'$(TargetFramework)' == 'net4.8'">
+    <AdditionalFiles Include="net4.8/PublicAPI.Shipped.txt" />
+    <AdditionalFiles Include="net4.8/PublicAPI.Unshipped.txt" />
+  </ItemGroup>
```

Comment:
Isn't this normally `net48` rather than `net4.8`?

Reply:
Also, I have this working:

```xml
    <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Shipped.txt" />
    <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Unshipped.txt" />
```

Reply:
@jnm2 Mind submitting a PR to update the documentation? Thanks!

Reply:
Sure!

Reply:
https://github.com/dotnet/roslyn-analyzers/pull/3447

Reply:
Both `net4.8` and `net48` work. Being explicit with the `.` will be important once we get to .NET 10, so I tend to add it in now, just to be clear.

Reply:
I hear you won't even have a choice when we get to `net5.0`, potentially.